openassessment/xblock/static/js/src/lms/components/WaitingStepDetailsPage.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Alert, Container, Row, Spinner,
+} from '@edx/paragon';
+import WaitingStepList from './WaitingStepList';
+
+const WaitingStepPage = ({ waitingStepDataUrl }) => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [waitingStepDetails, setWaitingStepDetails] = useState({
+    display_name: '', must_be_graded_by: '', must_grade: '', student_data: [],
+  });
+
+  const fetchWaitingStepDetails = async () => {
```

Comment:
Not a fan of network requests happening inside of component code...

Reply:
@muselesscreator I've pushed the request code to a file in the `api` folder.